editor/editor_properties.cpp

```diff
@@ -2810,7 +2810,11 @@ void EditorPropertyNodePath::_node_assign() {
 	if (!scene_tree) {
 		scene_tree = memnew(SceneTreeDialog);
 		scene_tree->get_scene_tree()->set_show_enabled_subscene(true);
-		scene_tree->set_valid_types(valid_types);
+		if (!valid_types.is_empty()) {
+			scene_tree->set_valid_types(valid_types);
+			scene_tree->get_scene_tree()->set_node_type_color_enabled(true);
+		}
```

Comment:
This change is wholly unnecessary I'd say. Instead you can make `set_node_item_type_color()` return early if valid types are empty.